### PR TITLE
Adding missing `Q` methods: `check()`, `flatten()`

### PR DESCRIPTION
**Pull request by Alexerson**
I noticed that 2 methods on the `Q` object were never typed.
I’m not 100% sure about the signature for `flatten`.

### CODE REVIEW

File: django-stubs/db/models/query_utils.pyi [hunk 1]
```diff
@@ -40,6 +40,8 @@ class Q(tree.Node):
         summarize: bool = ...,
         for_save: bool = ...,
     ) -> WhereNode: ...
+    def flatten(self) -> Iterator[Any]: ...
+    def check(self, against: dict[str, Any], using: str) -> bool: ...
```
Comment (Viicos) [inline django-stubs/db/models/query_utils.pyi]:
```suggestion
    def check(self, against: dict[str, Any], using: str = ...) -> bool: ...
```

File: django-stubs/db/models/query_utils.pyi [hunk 2]
```diff
@@ -40,6 +40,8 @@ class Q(tree.Node):
         summarize: bool = ...,
         for_save: bool = ...,
     ) -> WhereNode: ...
+    def flatten(self) -> Iterator[Any]: ...
```
Comment (Viicos) [inline django-stubs/db/models/query_utils.pyi]:
```suggestion
    def flatten(self) -> Iterator[Node | NothingNode | ???]: ...
```
Not sure if anything else could be yielded

**Review by Viicos — CHANGES_REQUESTED** [review]
You can also remove the corresponding entries from `scripts/stubtest/allowlist_todo.txt`

**Comment by Alexerson** [comment]
I fixed `allowlist_todo` and the signature for `check`.
As I said, not quite sure about the signature. I feel your suggestion might be good enough. If noone else suggest something else, I’ll include it.

**Comment by Viicos** [comment]
Yes I'm not sure either, `Any` would be sufficient if we can't really find what can be returned here.

File: django-stubs/db/models/query_utils.pyi (hunk 2 above)
Comment (intgr) [reply django-stubs/db/models/query_utils.pyi]: Our typing of `Node.children` and `Q.__init__(args, kwargs)` is incomplete also; `_NodeChildren` looks wrong when I look at `flatten()` code.

If you're not interested in un-entangling those, I think best to use `Incomplete` with `from _typeshed import Incomplete`

**Review by intgr — APPROVED** [review]
Thanks!

**Pull request merged**